pkg/image/importer/client_test.go

```diff
@@ -211,18 +211,93 @@ func TestDockerV1Fallback(t *testing.T) {
 func TestPing(t *testing.T) {
 	retriever := NewContext(http.DefaultTransport, http.DefaultTransport).WithCredentials(NoCredentials).(*repositoryRetriever)
 
-	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+	var fn http.HandlerFunc
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		if fn != nil {
+			fn(w, r)
+		}
+	})
+	server := httptest.NewServer(mux)
+	defer server.Close()
+
 	uri, _ := url.Parse(server.URL)
 
-	_, err := retriever.ping(*uri, true, retriever.context.InsecureTransport)
-	if !strings.Contains(err.Error(), "does not support v2 API") {
-		t.Errorf("Expected ErrNotV2Registry, got %v", err)
+	testCases := []struct {
+		name        string
+		uri         url.URL
+		expectNotV2 bool
```

Comment:
It's better to avoid negative names for boolean variables. I'd like to rename it into `expectV2`.